course5/homework12/comparers.h

```diff
@@ -0,0 +1,83 @@
+#pragma once
+#include <string>
+#include <cstring>
+#include <cstdio>
+#include <stdexcept>
+#include <boost/tokenizer.hpp>
+#include <memory>
+
+namespace homework12{
+
+class AComparer
+{
+public:
+    bool virtual operator() (const std::string& s1, const std::string& s2) const = 0;
```

Comment:
You are not using this class as a functor; instead, you wrap it in a lambda. Why using an overloaded function call operator then? I think simple function, like ```virtual bool Compare(a, b) const``` would be clearer.